e2e/vmss.go

```diff
@@ -55,6 +55,34 @@ func getNewRSAKeyPair(r *mrand.Rand) (privatePEMBytes []byte, publicKeyBytes []b
 func createVMSSWithPayload(ctx context.Context, customData, cseCmd, vmssName string, publicKeyBytes []byte, opts *scenarioRunOpts) (*armcompute.VirtualMachineScaleSet, error) {
 	model := getBaseVMSSModel(vmssName, opts.suiteConfig.location, *opts.chosenCluster.Properties.NodeResourceGroup, opts.subnetID, string(publicKeyBytes), customData, cseCmd)
 
+	isAzureCNI, err := opts.isChosenClusterAzureCNI()
+	if err != nil {
+		return nil, fmt.Errorf("failed to determine whether chosen cluster uses Azure CNI from cluster model: %s", err)
+	}
+
+	if isAzureCNI {
```

Comment:
could we move this logic out to a separate function, which would explain what's going on exactly?
Just for readability 😃